internal/serve/httphandler/wallets_handler.go

```diff
@@ -89,10 +90,41 @@ func (c WalletsHandler) DeleteWallet(rw http.ResponseWriter, req *http.Request)
 			httperror.NotFound("", err, nil).Render(rw)
 			return
 		}
-
 		httperror.InternalError(ctx, "", err, nil).Render(rw)
 		return
 	}
 
 	httpjson.RenderStatus(rw, http.StatusNoContent, nil, httpjson.JSON)
 }
+
+func (h WalletsHandler) PatchWallets(rw http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+
+	var reqBody *validators.PatchWalletRequest
+	if err := httpdecode.DecodeJSON(req, &reqBody); err != nil {
+		httperror.BadRequest("", err, nil).Render(rw)
+		return
+	}
+
+	validator := validators.NewWalletValidator()
+	validator.ValidatePatchWalletRequest(reqBody)
+	if validator.HasErrors() {
+		httperror.BadRequest("invalid request body", nil, validator.Errors).Render(rw)
+		return
+	}
+
+	walletID := chi.URLParam(req, "id")
+
+	err := h.Models.Wallets.SetEnabled(ctx, walletID, *reqBody.Enabled)
+	if err != nil {
+		if errors.Is(err, data.ErrRecordNotFound) {
+			httperror.NotFound("", err, nil).Render(rw)
+			return
+		}
+		err = fmt.Errorf("setting wallet enabled status: %w", err)
+		httperror.InternalError(ctx, "", err, nil).Render(rw)
+		return
+	}
+
+	httpjson.Render(rw, map[string]string{"message": "wallet enabled status updated successfully"}, httpjson.JSON)
```

Comment:
Since this method is a general PATCH, I think the result could be more generic in saying that the wallet was updated successfully.